examples/io_perf_test.cpp

```diff
@@ -0,0 +1,225 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Contributors to the OpenTimelineIO project
+
+#include <iostream>
+#include <cstdio>
+
+#include "opentimelineio/clip.h"
+#include "opentimelineio/typeRegistry.h"
+#include "util.h"
+#include <opentimelineio/any.h>
+#include <opentimelineio/serialization.h>
+#include <opentimelineio/deserialization.h>
+#include <opentimelineio/timeline.h>
+
+namespace otio = opentimelineio::OPENTIMELINEIO_VERSION;
+
+using chrono_time_point = std::chrono::steady_clock::time_point;
+
+const struct {
+    bool PRINT_CPP_VERSION_FAMILY    = false;
+    bool TO_JSON_STRING              = true;
+    bool TO_JSON_STRING_NO_DOWNGRADE = true;
+    bool TO_JSON_FILE                = true;
+    bool TO_JSON_FILE_NO_DOWNGRADE   = true;
+    bool CLONE_TEST                  = true;
+    bool SINGLE_CLIP_DOWNGRADE_TEST  = true;
+} RUN_STRUCT ;
+
+/// utility function for printing std::chrono elapsed time
+void
+print_elapsed_time(
+        const std::string& message,
+        const chrono_time_point& begin,
+        const chrono_time_point& end
+)
+{
+    const auto dur = (
+            std::chrono::duration_cast<std::chrono::microseconds>(
```

Comment:
If I cast to seconds then I don't get sub-second resolution, but maybe I'm doing it wrong?

```
downgrade clip: 0 [s]
deserialize_json_from_file: 1 [s]
serialize_json_to_string: 1 [s]
serialize_json_to_string [no downgrade]: 0 [s]
serialize_json_to_file: 0 [s]
serialize_json_to_file [no downgrade]: 0 [s]
```

this is what I tried:

```c++
    const auto dur = (
            std::chrono::duration_cast<std::chrono::seconds>(
                end - begin
            ).count()
    );
    std::cout << message << ": " << dur << " [s]" << std::endl;
```

Reply:
I'm not 100% sure, but I think it might be because `end` and `begin` are integers so subtracting them gives std::chrono::duration<int>.  If you try `std::chrono::duration<float>( end - begin)` that might do the trick.

Reply:
Got this working, thanks.